src/Aspire.Dashboard/Otlp/Security/ListenOptionsOtlpExtensions.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+
+namespace Aspire.Dashboard.Otlp.Security;
+
+internal static class ListenOptionsOtlpExtensions
+{
+    public static void UseOtlpConnection(this ListenOptions listenOptions)
+    {
+        listenOptions.Use(next => new OtlpConnectionMiddleware(next).OnConnectionAsync);
```

Comment:
Why not do it all inline in this single file?

Reply:
I can never remember connection middleware so I copied UseConnectionLogging and friends as a starting point.